carla/models/api/mlmodel.py

```diff
@@ -24,7 +24,13 @@ def __init__(
 
         if encoding_method == "OneHot":
             fitted_encoder = preprocessing.OneHotEncoder(
-                handle_unknown="ignore", sparse=False
+                handle_unknown="error", sparse=False
+            ).fit(data.raw[data.categoricals])
+            self.encoder: BaseEstimator = fitted_encoder
+
+        if encoding_method == "Binary":
+            fitted_encoder = preprocessing.OneHotEncoder(
+                drop="if_binary", handle_unknown="error", sparse=False
```

Comment:
We could, but if you then have more other encoders besides OneHotEncoder, and some arguments call different encoders while others set flags it might be more messy. Although I guess this is kinda duplicate code now.

Reply:
You could have something like

```
if encoding_method == "OneHot":
  drop = None
elif encoding_method == "Binary":
  drop = "if_binary
else:
  rase ...

fitted_encoder = preprocessing.OneHotEncoder(drop=drop, ...)
```

Then we would loose the duplicate code

Reply:
Oke, I guess we won't really use other encoding methods anyway.

Reply:
I saw you made an edit to this already, but didn't use the flags. I just kept your changes, so no flag for now.